js/plugins/drupalurl/plugin.js

```diff
@@ -0,0 +1,264 @@
+          editor.execCommand('editdrupalurl');
+        }
+      });
+    }
+  });
+
+  /**
+   * Get the surrounding drupalurl widget element.
+   *
+   * @param {CKEDITOR.editor} editor
+   */
+  function getSelectedEntity(editor) {
+    var selection = editor.getSelection();
+    var selectedElement = selection.getSelectedElement();
+    if (isEntityWidget(editor, selectedElement)) {
+      return selectedElement;
+    }
+
+    return null;
+  }
+
+  /**
+   * Returns whether or not the given element is a drupalurl widget.
+   *
+   * @param {CKEDITOR.editor} editor
+   * @param {CKEDITOR.htmlParser.element} element
+   */
+  function isEntityWidget (editor, element) {
+    var widget = editor.widgets.getByElement(element, true);
+    return widget && widget.name === 'drupalurl';
+  }
+
+  /**
+   * Generates unique HTML IDs for the widgets.
+   *
+   * @returns {string}
+   */
+  function generateEmbedId() {
+    if (typeof generateEmbedId.counter == 'undefined') {
+      generateEmbedId.counter = 0;
+    }
+    return 'url-embed-' + generateEmbedId.counter++;
+  }
+
+  /**
+   * Attaches or detaches behaviors, except the ones we do not want.
+   *
+   * @param {string} action
+   *   Either 'attach' or 'detach'.
+   * @param context
+   *   The context argument for Drupal.attachBehaviors()/detachBehaviors().
+   * @param settings
+   *   The settings argument for Drupal.attachBehaviors()/detachBehaviors().
+   */
+  function runEmbedBehaviors(action, context, settings) {
+    // Do not run the excluded behaviors.
+    var stashed = {};
+    $.each(Drupal.urlEmbed.excludedBehaviors, function (i, behavior) {
+      stashed[behavior] = Drupal.behaviors[behavior];
+      delete Drupal.behaviors[behavior];
+    });
+    // Run the remaining behaviors.
+    (action == 'attach' ? Drupal.attachBehaviors : Drupal.detachBehaviors)(context, settings);
+    // Put the stashed behaviors back in.
+    $.extend(Drupal.behaviors, stashed);
+  }
+
+  /**
+   * Ajax 'url_embed_insert' command: insert the rendered entity.
+   *
+   * The regular Drupal.ajax.commands.insert() command cannot target elements
+   * within iframes. This is a skimmed down equivalent that works whether the
+   * CKEditor is in iframe or divarea mode.
+   */
+  Drupal.AjaxCommands.prototype.url_embed_insert = function(ajax, response, status) {
+    var $target = ajax.element;
+    // No need to detach behaviors here, the widget is created fresh each time.
+    $target.html(response.html);
+    runEmbedBehaviors('attach', $target.get(0), response.settings || ajax.settings);
+  };
+
+  /**
+   * Stores settings specific to Entity Embed module.
+   */
+  Drupal.urlEmbed = {
+    /**
+     * A list of behaviors which are to be excluded while attaching/detaching.
+     *
+     * - Drupal.behaviors.editor, to avoid CK inception.
+     * - Drupal.behaviors.contextual, to keep contextual links hidden.
+     */
+    excludedBehaviors: ['editor', 'contextual']
```

Comment:
Not sure, but I think we can exclude these, since our output is not rendered entities, we won't have any issues with editor or contextual links in output.